presto-spi/src/main/java/com/facebook/presto/spi/block/AbstractMapBlock.java

```diff
@@ -211,7 +214,8 @@ public long getPositionsSizeInBytes(boolean[] positions)
         return getRawKeyBlock().getPositionsSizeInBytes(entryPositions) +
                 getRawValueBlock().getPositionsSizeInBytes(entryPositions) +
                 (Integer.BYTES + Byte.BYTES) * (long) usedPositionCount +
-                Integer.BYTES * HASH_MULTIPLIER * (long) usedEntryCount;
+                Integer.BYTES * HASH_MULTIPLIER * (long) usedEntryCount +
```

Comment:
hashTables.getRetainedSizeInBytes() ?

Reply:
hashTables.getRetainedSizeInBytes() is for retained size and accounts for the hashtable int[] array overhead 16 bytes, but calculateSize() is not supposed to account this. So